### PR TITLE
Add checking compatible types for Gathers GroupedGatherElimination

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Add checking compatible element types for `Gathers` indices input in `GroupedGatherElimination` transformation to avoid throw exception by `Concat` operator and not apply transformation.

### Tickets:
 - [CVS-123973](https://jira.devtools.intel.com/browse/CVS-123973)
 - [CVS-124939](https://jira.devtools.intel.com/browse/CVS-124939)

### CODE REVIEW

**Review by nikita-savelyevv — APPROVED** [review]
I confirm that the fix resolves the issue with red-pajama-3b-chat model compilation

File: src/common/transformations/src/transformations/common_optimizations/simplify_shape_of_sub_graph.cpp [hunk 1]
```diff
@@ -47,10 +47,11 @@ pass::GroupedGatherElimination::GroupedGatherElimination() {
                 continue;
             }
 
-            // Scalar inputs are not supported by Concat and we don't want to throw an exception here.
+            // Scalar or not same type inputs are not supported by Concat and we don't want to throw an exception here.
             // The transformation should not be applied instead.
             if (curr->input_value(1).get_partial_shape().same_scheme(Shape{}) ||
-                next->input_value(1).get_partial_shape().same_scheme(Shape{})) {
+                next->input_value(1).get_partial_shape().same_scheme(Shape{}) ||
+                !curr->input_value(1).get_element_type().compatible(next->input_value(1).get_element_type())) {
```
Comment (jane-intel) [inline src/common/transformations/src/transformations/common_optimizations/simplify_shape_of_sub_graph.cpp:54]: lets convert indices to same type? it looks reasonable

Comment (praasz) [reply src/common/transformations/src/transformations/common_optimizations/simplify_shape_of_sub_graph.cpp:54]: Ok, will check it.

**Comment by jane-intel** [comment]
@praasz looks like this PR solves bug CVS-124939. Please check via triggering hub tests

**Pull request merged**